packages/editor/src/editor/browser/extract/ExtractComponentDialog.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  BasicField,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+  Flex,
+  Input,
+  Message,
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  toast,
+  type BrowserNode
+} from '@axonivy/ui-components';
+import type { Component, ComponentData, Layout, Variable } from '@axonivy/form-editor-protocol';
+import { useEffect, useMemo, useState, type Dispatch, type ReactNode, type SetStateAction } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useAppContext } from '../../../context/AppContext';
+import { useQueryClient } from '@tanstack/react-query';
+import { genQueryKey } from '../../../query/query-client';
+import { useFunction } from '../../../context/useFunction';
+import { useMeta } from '../../../context/useMeta';
+import { variableTreeData } from '../data-class/variable-tree-data';
+import { useExtractFieldValidation } from './useExtractFieldValidation';
+type ExtractComponentDialogProps = {
+  children: ReactNode;
+  data: Component | ComponentData;
+  openDialog: boolean;
+  setOpenDialog: Dispatch<SetStateAction<boolean>>;
+};
+
+export const ExtractComponentDialog = ({ children, data, openDialog, setOpenDialog }: ExtractComponentDialogProps) => {
```

Comment:
as soon as the quick action bar opens, this code runs right? but this should only run if this dialog is really open. 
Split the dialog from the dialog content.